src/tlo/events.py

```diff
@@ -63,8 +65,54 @@ def apply(self, target):
 
     def run(self):
         """Make the event happen."""
+        
+        print_chains = False
+        df_before = []
+        
+        if self.sim.generate_event_chains:
+            # Only print event if it belongs to modules of interest and if it is not in the list of events to ignore
+            if (self.module in self.sim.generate_event_chains_modules_of_interest) and all(sub not in str(self) for sub in self.sim.generate_event_chains_ignore_events):
+                print_chains = True
+                if self.target != self.sim.population:
+                    row = self.sim.population.props.iloc[[self.target]]
```

Comment:
I believe it should be `.loc` rather `.iloc`. (The latter is used for ordinal position in the index, whereas we are using person_id which is the value in the index.) If we end-up doing memory-saving things like recycling the rows of dead persons for new persons, then the ordinal position in the index and index value won't the same any more.